### PR TITLE
replace cockpit.service with cockpit quadlet

### PR DESCRIPTION
**Pull request by Nitrousoxide**

### CODE REVIEW

**Review by bsherman — CHANGES_REQUESTED** [review]
Would you please change `ws` to `cockpit-ws` here?
```
ContainerName=ws
Environment=NAME=ws
```

It probably doesn't matter, but attempting minimize change as current container is `cockpit-ws`.

Let me know if that's undesirable for some reason.

**Comment by bsherman** [comment]
I have written in the README that `cockpit` does not start by default... This certainly does.

Now I don't know if my original README was wrong or not.

Regardless, can we even tell a quadlet to be disabled by default?

**Comment by bsherman** [comment]
This doesn't seem promising: https://docs.podman.io/en/latest/markdown/podman-systemd.unit.5.html#enabling-unit-files

File: main/etc/systemd/system/cockpit.service [hunk 1]
```diff
@@ -1,15 +1,34 @@
 [Unit]
-Description=quay.io/cockpit/ws:latest podman container
+Description=Cockpit Container
 After=network-online.target
 Wants=network-online.target
+SourcePath=/etc/containers/systemd/cockpit.container
```
Comment (bsherman) [inline main/etc/systemd/system/cockpit.service:5]: Tested and this `SourcePath=` line needs to go away. It's technically wrong now that we removed the container file, but more importantly, it results in systemd constantly warning.

```
systemctl status cockpit
Warning: The unit file, source configuration file or drop-ins of cockpit.service changed on disk. Run 'systemctl daemon-reload' to reload units.
```

File: main/etc/systemd/system/cockpit.service [hunk 2]
```diff
@@ -1,15 +1,34 @@
...
+Restart=always
+Environment=PODMAN_SYSTEMD_UNIT=%n
+KillMode=mixed
+ExecStopPost=-/usr/bin/podman rm -f -i --cidfile=%t/%N.cid
+ExecStopPost=-rm -f %t/%N.cid
+Delegate=yes
+Type=notify
+NotifyAccess=all
+SyslogIdentifier=%N
+ExecStart=/usr/bin/podman run --name=cockpit-ws --cidfile=%t/%N.cid --replace --rm --log-driver passthrough --cgroups=split --sdnotify=conmon -d ->
```
Comment (bsherman) [inline main/etc/systemd/system/cockpit.service:31]: @Nitrousoxide 
the `ExecStart=`  was failing for me in testing this too... I think i see what happened  `-d ->` looks like the output got cut off by `systemctl cat` not line-wrapping.

Correct line should be:
```
ExecStart=/usr/bin/podman run --name=ws --cidfile=%t/%N.cid --replace --rm --cgroups=split --sdnotify=conmon -d -v /:/host --env NAME=ws --label io.containers.autoupdate=registry --pid host --privileged quay.io/cockpit/ws:latest /container/label-run
```

File: main/etc/systemd/system/cockpit.service (hunk 1 above)
Comment (Nitrousoxide) [reply main/etc/systemd/system/cockpit.service]: i pushed this fix.  It does start up in my vm and can auto-update though the logs do throw some errors on the container spinning up:

```
May 16 03:44:45 coreos-cockpit-02 ws[3019]: + mkdir -p /host/etc/cockpit/ws-certs.d /host/etc/cockpit/machines.d
May 16 03:44:45 coreos-cockpit-02 ws[3019]: + chmod 755 /host/etc/cockpit/ws-certs.d /host/etc/cockpit/machines.d
May 16 03:44:45 coreos-cockpit-02 ws[3019]: + chown root:root /host/etc/cockpit/ws-certs.d /host/etc/cockpit/machines.d
May 16 03:44:45 coreos-cockpit-02 ws[3019]: + mkdir -p /etc/ssh
May 16 03:44:45 coreos-cockpit-02 ws[3019]: + '[' podman = oci -o podman = podman ']'
May 16 03:44:45 coreos-cockpit-02 ws[3019]: + '[' -n '' ']'
May 16 03:44:45 coreos-cockpit-02 ws[3019]: + /bin/mount --bind /host/etc/cockpit /etc/cockpit
May 16 03:44:45 coreos-cockpit-02 ws[3019]: + /usr/libexec/cockpit-certificate-ensure
May 16 03:44:45 coreos-cockpit-02 ws[3019]: '/etc/os-release' -> '/host/etc/os-release'
May 16 03:44:45 coreos-cockpit-02 ws[3019]: '/usr/lib/os-release' -> '/host/usr/lib/os-release'
```

which seem to be due to the startup script in "/container/label-run" inside the container

Comment (bsherman) [reply main/etc/systemd/system/cockpit.service]: Yep, i see the same on my VM. I don't think it's errors, just verbose `bash -x` type output.

**Pull request merged**